src/hooks/useForm.ts

```diff
@@ -0,0 +1,84 @@
+import { ChangeEvent, useEffect, useState } from 'react'
+import { emailRegex, nameRegex, requiredRegex } from '../texts/errors/formErrors'
+
+// input data types supported
+export type typeTypes = 'text' | 'name' | 'email'
+
+// taken and modified from https://learnetto.com/blog/react-form-validation
+const useForm = (defaultValues: Record<string, string>) => {
+  // Form values
+  const [values, setValues] = useState<Record<string, string>>(defaultValues)
+  // Errors
+  const [errors, setErrors] = useState<Set<string>>(new Set())
```

Comment:
any rationale for using set / record? would a map work better, or a simple array? 

If you're working with an API on a scalable feature (e.g. supper ordering), would you want your data structure to be immutable?


Don't mind me if it's been considered, just poking some thoughts around for the learning.